test/utils.test.ts

```diff
@@ -117,10 +123,238 @@ describe('utils', function () {
         .onSecondCall()
         .resolves(availableRelayHub);
       const {
-        relayInfo: { url: actualUrl },
+        managerData: { url: actualUrl },
       } = (await selectNextRelay(httpClientStub)) as RelayInfo;
 
       expect(actualUrl).to.equal(expectedUrl);
     });
   });
+
+
+  describe('validateRelayResponse', function () {
+    let deployRequest: DeployTransactionRequest;
+    let relayWorkerAddress: string;
+    let relayHubAddress: string;
+
+    beforeEach(function () {
+      relayWorkerAddress = createRandomAddress();
+      relayHubAddress = FAKE_ENVELOPING_CONFIG.relayHubAddress;
+
+      deployRequest = {
+        metadata: {
+          relayHubAddress,
+          relayMaxNonce: 6,
+          signature: '',
+        },
+        relayRequest: {
+          relayData: {
+            callForwarder: '',
+            callVerifier: '',
+            feesReceiver: '',
+            gasPrice: '',
+          },
+          request: {
+            from: '',
+            to: '',
+            index: 0,
+            nonce: 0,
+            recoverer: createRandomAddress(),
+            relayHub: '',
+            tokenAmount: 0,
+            tokenContract: '',
+            tokenGas: 0,
+            value: 0,
+            data: '',
+          },
+        },
+      };
```

Comment:
I'd suggest to use the existing fakes, or at least define fakes with "thruthly" values, with integers > 1 to make sure that calculations are performed correctly.

Reply:
updated